sdk/eventhub/azure-eventhub/tests/livetest/asynctests/test_send_async.py

```diff
@@ -529,3 +539,49 @@ async def on_success(events, pid):
         assert sent_events[-1][1] == "0"
 
         assert not on_error.err
+
+@pytest.mark.parametrize("keep_alive", [None, 30, 60])
+@pytest.mark.liveTest
+@pytest.mark.asyncio
+def test_send_with_keep_alive_async(connstr_receivers, keep_alive, uamqp_transport):
+    connection_str, receivers = connstr_receivers
+    client = EventHubProducerClient.from_connection_string(connection_str, keep_alive=keep_alive, uamqp_transport=uamqp_transport)
+    assert client._producers["all-partitions"]._keep_alive == keep_alive
+
+
+@pytest.mark.parametrize("keep_alive", [None, 5, 30])
+@pytest.mark.liveTest
+@pytest.mark.asyncio
+async def test_send_long_wait_idle_timeout(connstr_receivers, keep_alive, uamqp_transport):
+    if uamqp_transport:
+        amqp_transport = UamqpTransport
+        retry_total = 3
+    else:
+        amqp_transport = PyamqpTransport
+        retry_total = 0
+    connection_str, receivers = connstr_receivers
+    client = EventHubProducerClient.from_connection_string(connection_str, keep_alive=keep_alive, idle_timeout=10, retry_total=retry_total, uamqp_transport=uamqp_transport)
+    sender = await client._create_producer(partition_id="0")
+    async with sender:
+        await sender._open_with_retry()
+        ed = EventData('data')
+        ed = transform_outbound_single_message(ed, EventData, amqp_transport.to_outgoing_amqp_message)
+        sender._unsent_events = [ed._message]
+        # hit idle timeout error
+        await asyncio.sleep(11)
+
+        if uamqp_transport:
+            sender._unsent_events[0].on_send_complete = sender._on_outcome
+            if keep_alive !=5:
+                with pytest.raises((uamqp.errors.ConnectionClose,
+                                        uamqp.errors.MessageHandlerError, OperationTimeoutError)):
+                    await sender._send_event_data()
+            else:
+                await sender._send_event_data()
+
+        if not uamqp_transport:
+            if keep_alive == 5:
```

Comment:
just checking,  this should be `if keep_alive` for case 5 and 30 ?

Reply:
I think for 30, it will still error out since idle_timeout is 10, so it should go into the else block.

Reply:
yup! just for 5 :)